examples/browser-webrtc/server/src/lib.rs

```diff
@@ -0,0 +1,138 @@
+    )
+    .map(|(peer_id, conn), _| (peer_id, StreamMuxerBox::new(conn)))
+    .boxed();
+
+    let behaviour = Behaviour {
+        relay: relay::Behaviour::new(local_peer_id, Default::default()),
+        ping: ping::Behaviour::new(ping::Config::new()),
+        keep_alive: keep_alive::Behaviour,
+    };
+
+    let mut swarm = SwarmBuilder::with_tokio_executor(transport, behaviour, local_peer_id).build();
+
+    let address_webrtc = Multiaddr::from(Ipv6Addr::LOCALHOST)
+        .with(Protocol::Udp(0))
+        .with(Protocol::WebRTCDirect);
+
+    swarm.listen_on(address_webrtc.clone())?;
+
+    // Dial the peer identified by the multi-address given as the second
+    // command-line argument, if any.
+    if let Some(addr) = remote {
+        log::info!("Dialing {addr}");
+        swarm.dial(addr)?;
+    }
+
+    let mut addr = None; // We only need 1 address
+
+    loop {
+        tokio::select! {
+            evt = swarm.select_next_some() => {
+                match evt {
+                    SwarmEvent::NewListenAddr { address, .. } if addr.is_none() => {
+
+                        addr = Some(address
+                                .with(Protocol::P2p(*swarm.local_peer_id()))
+                                .clone()
+                                .to_string());
+
+                        let address = addr.as_ref().unwrap().clone();
+
+                        tokio::spawn(async move {
+
+                            log::info!("Serving the Multiaddr we are listening on: {}", address);
+
+                            let app = Router::new().route("/", get(|| async { address }))
+                            .layer(
+                                // allow cors
+                                CorsLayer::new()
+                                    .allow_origin(Any)
+                                    .allow_methods([Method::GET]),
+                             );
+
+                            axum::Server::bind(&SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), PORT))
+                                .serve(app.into_make_service())
+                                .await
+                                .unwrap();
+                        });
+
+                        log::debug!("Server spawned");
+                    }
+                    SwarmEvent::Behaviour(Event::Ping(ping::Event {
+                        peer,
+                        result: Ok(rtt),
+                        ..
+                    })) => {
+                        let id = peer.to_string().to_owned();
+                        log::info!("🏓 Pinged {id} ({rtt:?})")
+                    }
+                    evt => {
+                        log::debug!("SwarmEvent: {:?}", evt);
+                    },
+                }
+            },
+            _ = tokio::signal::ctrl_c() => {
+                break;
+            }
+        }
+    }
+    Ok(())
+}
+
+#[derive(NetworkBehaviour)]
+#[behaviour(to_swarm = "Event", prelude = "libp2p_swarm::derive_prelude")]
```

Comment:
Any particular reason for using `to_swarm` here and not have the event auto-generated? Would save you a few LoC.

Reply:
No reason other than what I must have seen somewhere and thought it was necessary. Is there a macro for this or is it already part of `#[derive(NetworkBehaviour)]`?

Reply:
I figured it out. 

Honestly, the whole `#[derive(NetworkBehaviour)]` process has always been a bit confusing and the onus is on the user to know all the moving parts: Developers have to remember to include the "macros" feature in libp2p, remember to `use` the `NetworkBehaviour` and derive it, and know that the event is called `BehaviourEvent`. Beyond the scope here, but thought I'd mention it as there may be others struggling to wrap their heads around the process.